### PR TITLE
Fixing issue #679 - Feature/feed filters posts endpoint

### PR DESCRIPTION
**Pull request by CeesWang**
### What does this pull request aim to achieve? 💯

_Please be concise and link any related issue(s):_

### 🚨Before submitting this pull request🚨:

_Please do **NOT** submit this PR if you have not done the following:_

1. I have checked that no one else is working on similar changes.
2. I have read the latest [**README**](https://github.com/FightPandemics/FightPandemics/blob/master/README.md) and understand the development workflow.
3. The name of this branch is: **`feature/<branch_name>`** (need to have cloned the repo not forked).
4. This branch is rebased/merged with the **latest master**.
5. There are no merge conflicts.
6. There are no console warnings and errors.
7. On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".

### CODE REVIEW

File: client/src/pages/Feed.js [hunk 1]
```diff
@@ -317,14 +328,35 @@ const Feed = (props) => {
...
-    const { user } = props;
     const limit = 5;
     const skip = page * limit;
-    /* Add userId when user is logged */
-    const endpoint = `/api/posts?limit=${limit}&skip=${skip}${
-      user && user.userId ? `&userId=${user.userId}` : ""
-    }`;
+    const baseURL = `/api/posts?limit=${limit}&skip=${skip}`;
+    let endpoint = `${baseURL}${objectiveURL()}${filterURL()}`;
+    console.log(endpoint);
```
Comment (idjevm) [inline client/src/pages/Feed.js:359]: Do we need to have this console log? Should be removed.

File: client/src/assets/data/filterOptions.js [hunk 2]
```diff
@@ -17,15 +17,15 @@ export default {
     ],
   },
   type: {
-    label: "Type",
+    label: "type",
     className: "filter-3",
     options: [
-      "Medical supplies",
-      "Groceries / Food",
```
Comment (idjevm) [inline client/src/assets/data/filterOptions.js:24]: Will the filter not work if you have the spaces between the / ? If they will, I think we should still have this for better presentation.

File: client/src/assets/data/filterOptions.js [hunk 3]
```diff
@@ -17,15 +17,15 @@ export default {
...
+      "Medical Supplies",
+      "Groceries/Food",
       "Business",
       "Education",
       "Legal",
-      "Wellbeing / Mental",
```
Comment (idjevm) [inline client/src/assets/data/filterOptions.js:28]: Same as the last comment about the /

File: client/src/assets/data/filterOptions.js (hunk 2 above)
Comment (joshmorel) [reply client/src/assets/data/filterOptions.js:24]: @idjevm They need to match the backend `POST_TYPES` as defined. Interestingly the original design has spaces for "Groceries / Food" but no spaces for "Wellbeing/Mental". But I would say this it's fine the data model was approved with these values.

File: package-lock.json [hunk 4]
```diff
@@ -2,10 +2,138 @@
   "requires": true,
   "lockfileVersion": 1,
   "dependencies": {
```
Comment (joshmorel) [inline package-lock.json:4]: This is weird - we have `package-lock.json` in root folder (we should only have in `backend/` & `client/`). Looks like it was removed and then added back (I guess accidentally). Anyways I would remove this change from the PR.

**Pull request merged**